### ISSUE TITLE
flux-reload command in the core

### ISSUE THREAD

**Issue by dongahn** [flux-framework/flux-core#501]
I now have some testing infrastructure set up within `flux-sched`, which makes use of hwloc reloading support. So far I have been using `sleep k` to wait for the reload to finish, but now I need to switch to the rpc-based reload. 

Initially, I thought I was going to implement a `flux_rpc_multi`-based frontend command within the `flux-sched,` but I realize this may actually better belong to `flux-core`, helping other projects beyond `flux-sched` set up their tests. This includes what will go into the [flux-scalability](https://github.com/flux-framework/flux-scalability), which I also plan to contribute to. 

If folks don't object, I can start to write a simple front-end command in `flux-core/src/cmd` with something like:

```
Usage: reload [OPTIONS] ARGS
    -r,--rankset RANKSET Send hwloc reload requests to brokers in RANKSET.
ARGS:
    HWLOC_XML_PATH       Directory path that contains valid per-rank hwloc
                         resource xml files named <rank>.xml. 
                         E.g., 0.xml and 1.xml must reside in HWLOC_XML_PATH files
                         for --rankset=0-1.
```

Thoughts?

**Comment by lipari** [comment]
`flux reload` seems a bit too ambiguous.  How about `flux resource reload` or `flux resource refresh` or `flux hwloc reload|refresh`?

**Comment by dongahn** [comment]
@lipari: 

Indeed I thought about two ways to create a higher level command: 1) making  resource-hwloc as a higher level command and have subcommands underneath, which includes reload -- this will be similar to your `flux resource reload`. 2) If we want to be even more generic, perhaps one can create a flux-rpc. 

The reason I suggested flux reload was simplicity as my current main focus is firming up testing infra. But I guess I can as easily create `flux resource reload` which then can later extended to cover other resource-hwloc request callbacks.

**Comment by dongahn** [comment]
Perhaps what I should do is make `reload` as a subcommand of flux-hwloc and implement only that subcommand.

```
Usage: flux-hwloc reload [OPTIONS] ARGS
       flux-hwloc query [OPTIONS]
       flux-hwloc get   [OPTIONS]

OPTIONS:
    -h,--help
    -r,--rankset RANKSET Send hwloc requests to brokers in RANKSET.

ARGS:
    HWLOC_XML_PATH       Directory path that contains valid per-rank hwloc
                         resource xml files named <rank>.xml. 
                         E.g., 0.xml and 1.xml must reside in HWLOC_XML_PATH files
                         for --rankset=0-1.
```

Because `lux-lstopo` already exists, `flux-hwloc topo` wouldn't be necessary. Also since `query` and `get` are to be implemented within `resource-hwloc` module, having these as a placeholder could facilitate the command line implementation when these callbacks are implemented?

@trws: are you okay with this? Or did you have other thoughts about a standalone command for a test to interact with the `resource-hwloc` module?

**Comment by trws** [comment]
That sounds fine, but if this is happening, it should probably be a `flux hwloc <subcommand>` structure so that lstopo can become part of the same thing, and we can add support for some of the other common hwloc query commands the same way.  actually, unless you have a PR for this I should probably just do it by re-working the python-based lstopo.

**Comment by dongahn** [comment]
@trws: that's what I exactly ended up doing, but in C. And the implementation is mostly ready. I will add some sharness test cases around it and make a separate PR -- I will appreciate your feedback then. 

Sorry I moved ahead a bit too quickly. But I needed it to make progress in my `sched` testing setup.

**Comment by trws** [comment]
Ok, if you've already got it ready then that should be great.  I was just thinking that making a command with subcommands is a great deal less work with argparse, because it's built for the purpose, and might be easier to get you started faster.

**Comment by dongahn** [comment]
Ok. This has been addressed with PR #506. Closing.
